contrib/mod_sftp/fxp.c

```diff
@@ -4384,14 +4384,13 @@ static int fxp_handle_ext_check_file(struct fxp_packet *fxp, char *digest_list,
     pr_signals_handle();
 
     res = BIO_read(bio, data, data_len);
-    if (res < 0) {
-      if (BIO_should_read(bio)) {
-        continue;
-      }
+    xerrno = errno;
 
-      /* error */
-      xerrno = errno;
+    pr_trace_msg(trace_channel, 19,
+      "reading %d bytes from '%s' via BIO yielded %d", data_len, path, res);
 
+    if (res <= 0) {
```

Comment:
Hmm.  What if the file size is legitimately zero bytes here?  I'll update this to add a check along these lines:
```
if (res < 0 ||
    (res == 0 && st.st_size != 0)) {
  /* error */
  ...
}
```